sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core.TestFramework
+{
+    public class ProxyTransport : HttpPipelineTransport
+    {
+        private readonly HttpPipelineTransport _innerTransport;
+        private readonly TestRecording _recording;
+        private readonly TestProxy _proxy;
+
+        public ProxyTransport(TestProxy proxy, HttpPipelineTransport transport, TestRecording recording)
+        {
+            if (transport is HttpClientTransport)
+            {
+                var handler = new HttpClientHandler
+                {
+                    ServerCertificateCustomValidationCallback = (_, certificate, _, _) => certificate.Issuer == TestProxy.DevCertIssuer
+                };
+                _innerTransport = new HttpClientTransport(handler);
+            }
+            // HttpWebRequestTransport
+            else
+            {
+                ServicePointManager.ServerCertificateValidationCallback += (_, certificate, _, _) => certificate.Issuer == TestProxy.DevCertIssuer;
```

Comment:
This will disable the cert check for everyone until the end of time, you might want to scope this to a single endpoint of test proxy